os/compression/compress.c

```diff
@@ -66,3 +86,41 @@ int compress_block(unsigned char *out_buffer, long unsigned int *writesize, unsi
 #endif
 	return ret;
 }
+
+/****************************************************************************
+ * Name: decompress_block
+ *
+ * Description:
+ *   Decompress block in 'read_buffer' of readsize into 'out_buffer' of writesize
+ *
+ * Returned Value:
+ *   Non-negative value on Success.
+ *   Negative value on Failure.
+ ****************************************************************************/
+ #if CONFIG_COMPRESSION_TYPE == LZMA
+int decompress_block(unsigned char *out_buffer, unsigned int *writesize, unsigned char *read_buffer, unsigned int *size)
+#elif CONFIG_COMPRESSION_TYPE == MINIZ
+int decompress_block(unsigned char *out_buffer, long unsigned int *writesize, unsigned char *read_buffer, long unsigned int *size)
```

Comment:
This looks inconvenient.
If we provide this API like this, application should keep the same conditional.
```
in app code

#if type == lzma
decompress_block(....);
#elif type == miniz
decompress_bloc(...);
#endif
```
It means this driver does not have a functionality of abstraction layer.
Could you make a way for single API regardless of compression type?
It should be applied in compression function as well.

Reply:
done